### DISCUSSION TITLE
<My Prettier extenion does not format the php file.>

### DISCUSSION THREAD

**Discussion by Habibsultani** [community/community#75011] [Programming Help] [answered]
### Select Topic Area

Question

### Body

Whenever I press the (ctr + s)  my prettier extension does not format the code in php file. while whenever I am working with other language like JavaScript the prettier is formatting the code correctly. So what should I do. Is there any extension for formating the php files. Or you suggest other IDE for writting php.

Thanks.

**Comment by ivnvxd** [comment, marked as answer]
Install [PHP Plugin](https://github.com/prettier/plugin-php) for Prettier: Prettier requires a plugin to format PHP files. You can install the prettier-plugin-php package. In your terminal, run:
`npm install --save-dev prettier prettier-plugin-php`
Ensure this plugin is referenced in your Prettier configuration file.

In VSCode settings, make sure that Prettier is set as the default formatter for PHP files. You can search for 'Default Formatter' and ensure it's set to Prettier for PHP.

**Reply by ivnvxd** [reply]
I'd like to ask, did this help solve your issue? If so, I would greatly appreciate it if you marked it as an answer.
